### PR TITLE
optimize to split lines

### LINKED ISSUE #1324: in_tail uses regexp to split lines
https://github.com/fluent/fluentd/blob/master/lib/fluent/plugin/in_tail.rb#L573

It's much slower than using `index` and `slice!`.
Here's a micro bench code and its result:
https://gist.github.com/tagomoris/d9cf1e961378ab553b65fbaa844ba6df

### PR DESCRIPTION
**Pull request by tagomoris**
fixes #1324.

### CODE REVIEW

**Comment by tagomoris** [comment]
@repeatedly Could you review this change?
Benchmark is much nice to have.

File: lib/fluent/plugin/in_tail.rb [hunk 1]
```diff
@@ -570,8 +570,8 @@ def on_notify
                   else
                     @buffer << @io.readpartial(2048, @iobuf)
                   end
-                  while line = @buffer.slice!(/.*?\n/m)
-                    @lines << line
+                  while idx = @buffer.index("\n")
```
Comment (repeatedly) [inline lib/fluent/plugin/in_tail.rb]: For more small optimization, `"\n".freeze` is better,

Comment (tagomoris) [inline lib/fluent/plugin/in_tail.rb]: Right.

**Comment by tagomoris** [comment]
I pushed an updated commit.

**Comment by repeatedly** [comment]
Looks good to me

**Pull request merged**